### PR TITLE
fix(kb): reject web ingest start_url without http/https scheme

### PR DESCRIPTION
**Pull request by yiboyasss**

### CODE REVIEW

File: src/xagent/web/api/kb.py [hunk 1]
```diff
@@ -2325,6 +2343,8 @@ async def ingest_web(
                 status_code=422, detail=f"Invalid collection name: {str(e)}"
             ) from e
 
+        start_url = _validate_web_start_url(start_url)
```
Comment (rogercloud) [inline src/xagent/web/api/kb.py]: The validation is currently added only in the `/ingest-web` endpoint, but URL validity is part of the shared web-crawl contract. The related logic is now split across `src/xagent/web/api/kb.py` and `URLFilter.normalize_url()`, so callers that construct `WebCrawlConfig` directly, such as `CreateKnowledgeBaseFromUrlTool`, bypass the new check and can still return a misleading `success` result with 0 documents for `www.example.com`.

There are also consistency gaps in the split logic: `HTTP://example.com` is accepted by the endpoint helper because it lower-cases the scheme, but `URLFilter.normalize_url()` still rejects it through a case-sensitive prefix check; and hostless URLs such as `http://@` / `http://:80` pass the helper's `netloc` check even though `parsed.hostname` is `None`, then fail later as crawler errors instead of 422 validation errors.

Can we consolidate the URL scheme/hostname validation and normalization in the shared crawl layer instead of keeping endpoint-specific checks? The concrete places I think need to be aligned are:

- `src/xagent/core/tools/core/RAG_tools/core/schemas.py`: make `WebCrawlConfig.start_url` validate/normalize the URL, or delegate to a shared helper.
- `src/xagent/core/tools/core/RAG_tools/web_crawler/url_filter.py`: use the same normalized URL semantics as the config validator, including case-insensitive schemes and real hostname checks.
- `src/xagent/web/api/kb.py`: have `/ingest-web` rely on that shared validation path rather than owning a separate helper.
- `src/xagent/core/tools/adapters/vibe/web_ingestion_tool.py`: this path should get the same behavior automatically when it builds `WebCrawlConfig`.
- Tests: cover the API path plus a direct/tool `WebCrawlConfig` path, including `www.example.com`, uppercase schemes like `HTTP://example.com`, and hostless URLs like `http://@`.

That keeps the related logic in one place and avoids partial fixes where one wrapper rejects bad URLs but another ingestion path still silently drops them.

**Review by rogercloud — COMMENTED** [review]
Found one validation-boundary issue around web-ingestion URL handling.

File: src/xagent/core/tools/core/RAG_tools/core/web_url_utils.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        userinfo = parsed.username
+        if parsed.password is not None:
+            userinfo = f"{userinfo}:{parsed.password}"
+        userinfo = f"{userinfo}@"
+
+    port = ""
+    if parsed.port is not None:
+        port = f":{parsed.port}"
+
+    return f"{userinfo}{hostname.lower()}{port}"
```
Comment (rogercloud) [inline src/xagent/core/tools/core/RAG_tools/core/web_url_utils.py]: This rebuilds `netloc` from `parsed.hostname`, but `urlparse()` strips the brackets from IPv6 literal hosts. A valid input such as `http://[::1]:8000/docs#frag` is normalized to `http://::1:8000/docs`, which downstream URL parsers reject as an invalid port. The old normalization path preserved `parsed.netloc`, so this is a regression introduced by the shared helper.

Can we preserve IPv6 brackets when rebuilding the host, e.g. wrap hosts containing `:` as `[host]` before appending the port, and add a regression test like `WebCrawlConfig(start_url="http://[::1]:8000/docs#frag").start_url == "http://[::1]:8000/docs"`?

**Review by rogercloud — COMMENTED** [review]
Found one remaining edge-case regression in URL normalization.

**Review by qinxuye — APPROVED** [review]
LGTM
